### ISSUE TITLE
[android][background-fetch] Background fetch does not execute the task

### ISSUE THREAD

**Issue by dhirajksharma** [expo/expo#33596]
### Minimal reproducible example

https://github.com/dhirajksharma/my-app

### What platform(s) does this occur on?

Android

### Where did you reproduce the issue?

in a development build

### Summary

expo-background-fetch is able to register a task defined using expo-task-manager but is not able to execute the task even after the `minimumInterval` has long passed. 

expo-doctor says that expo-background-fetch is not supported on the new architecture.

### Environment

```text
expo-env-info 1.2.1 environment info:
    System:
      OS: Linux 6.10 Fedora Linux 40 (Workstation Edition)
      Shell: 5.2.26 - /bin/bash
    Binaries:
      Node: 20.17.0 - /usr/bin/node
      Yarn: 1.22.22 - /usr/local/bin/yarn
      npm: 10.8.2 - /usr/bin/npm
    npmPackages:
      expo: ~52.0.18 => 52.0.18 
      expo-router: 4.0.11 => 4.0.11 
      react: 18.3.1 => 18.3.1 
      react-dom: 18.3.1 => 18.3.1 
      react-native: 0.76.5 => 0.76.5 
      react-native-web: ~0.19.13 => 0.19.13 
    npmGlobalPackages:
      eas-cli: 13.4.2
    Expo Workflow: managed
```

### Expo Doctor Diagnostics

```text
Enabled experimental React Native Directory checks. Unset the EXPO_DOCTOR_ENABLE_DIRECTORY_CHECK environment variable to disable this check.
✔ Check package.json for common issues
✔ Check Expo config for common issues
✔ Check native tooling versions
✔ Check if the project meets version requirements for submission to app stores
✔ Check for common project setup issues
✔ Check dependencies for packages that should not be installed directly
✔ Check for app config fields that may not be synced in a non-CNG project
✔ Check for issues with Metro config
✔ Check npm/ yarn versions
✔ Check Expo config (app.json/ app.config.js) schema
✖ Validate packages against React Native Directory package metadata
✔ Check that native modules do not use incompatible support packages
✔ Check for legacy global CLI installed locally
✔ Check that packages match versions required by installed Expo SDK
✔ Check that native modules use compatible support package versions for installed Expo SDK

Detailed check results:

The following issues were found when validating your dependencies against React Native Directory:
Unsupported on New Architecture: expo-background-fetch
Advice: 
- Use libraries that are actively maintained and support the New Architecture. Find alternative libraries with https://reactnative.directory.
- Add packages to expo.doctor.reactNativeDirectoryCheck.exclude in package.json to selectively skip validations, if the warning is not relevant.

One or more checks failed, indicating possible issues with the project.
```

**Comment by chrfalch** [comment]
Thanks, @dhirajksharma :)

**Comment by chrfalch** [comment]
`expo-background-fetch` should be working on the new arch - I'll add a PR to update React-native-directory.

In your example you're defining the task inside of a screen `app/(tabs)/index.tsx`. This might be a mistake since if the app is killed and started in the background this screen will not be imported and run. You should definitely start by defining it (or importing the index file) from the `app/_layout.tsx` file - which should be loaded on startup.

On Android, `expo-background-fetch` works by setting an alarm when the app is backgrounded. The alarm will trigger executing the background task.

I tested your repo (and  adjusted the interval down to 1 minute to make it trigger a bit faster) and after starting it, registering the task and then background the app - the task was called as expected.

If the app is killed the alarm won't be started - and hence no background task will be executed.

**Comment by chrfalch** [comment]
Added a PR to `react-native-community/directory` updating the newArch flag:

https://github.com/react-native-community/directory/pull/1420

This issue should be closed when this PR is merged.

**Comment by dhirajksharma** [comment]
> In your example you're defining the task inside of a screen `app/(tabs)/index.tsx`. This might be a mistake since if the app is killed and started in the background this screen will not be imported and run. You should definitely start by defining it (or importing the index file) from the `app/_layout.tsx` file - which should be loaded on startup.

I understand now, I did as per your instructions and the background task was executed as expected. But, it gets executed only once. As per the expo docs, it should be executed periodically right?
 
> If the app is killed the alarm won't be started - and hence no background task will be executed.

Again from the expo docs, it does [say](https://docs.expo.dev/versions/latest/sdk/background-fetch/#known-issues), that the BackgroundFetch will work as long as the app is in the background and not if it is killed. But it has only the ios label. Furthermore, when we set `stopOnTerminate` as `false` when calling `BackgroundFetch.registerTaskAsync()` is it not to continue the task execution upon app termination?

**Comment by chrfalch** [comment]
I'm reopening, after testing I could also see that background fetch was only run once. Will fix this with another PR - should definitely work like you describe.

When it comes to the docs I'll see if I can update there as well.

**Comment by chrfalch** [comment]
I was a bit quick here - I just let the app sit in the background without bringing it to the foreground, and I could verify that it was repeated. The scheduled alarm is a repeating alarm.

**Comment by chrfalch** [comment]
The task is supposed to be executed when the app is stopped (with the `stopOnTerminate` flag set to false as you do). 

Force stopping an app, on the other hand, (which happens on a lot of devices when the user swipes the app out of the recents screen) **will** clear all alarms. 

If the app crashes or is stopped by the system somehow, it should not clear alarms and the app should be restarted in the background and execute your background tasks.

There should be a lot of resources online that touches upon the difference between force stopping and closing Android apps, also multiple discussions about how some Android vendors do a force stop when the user removes an app from the recent apps list - which is unfortunately not something we're able to handle.

I'm testing this behavior now to see if it all works as expected. There as traditionally been some issues with the background tasks and force stopped apps that we're trying to fix.

**Comment by chrfalch** [comment]
I can verify that if I use the "Kill process" options from within the Logcat view in Android studio, the app is restarted and correctly executes the task in the background.

**Comment by dhirajksharma** [comment]
okay i understand it a bit more now. I will read further upon it as you suggest. Thanks @chrfalch you've been really helpful :)
